### DISCUSSION TITLE
Discussion about Application Flow

### DISCUSSION THREAD

**Discussion by rasyidf** [SkalskiP/make-sense#295] [General]
One of my concerns about **make sense** development is how the application flow and state managed.

Let me just list some of mine, here 😢:

- Every Hot Reload made will lead back to start, there's no way I can get the state back. 
- There's no cache management, so every model that have been pulled will not be reused after reload, or by different tab.
- The Application flow route start from landing, not the editor, it's fine for some cases, but this kind of SPA isn't.

**Comment by SkalskiP** [comment]
@rasyidf any ideas on how to solve those issues? :)

**Reply by rasyidf** [reply]
I'm  searching for the best approach for this, so far this might be a solution. 

1. we can use some routing management with state syncing middleware (I'm not sure how yet) react routing or react-location will probably help

2. we can use IndexedDB storage. and presist the models there. so it will be saved.
